### PR TITLE
Add aggregate requests contexts by request id

### PR DESCRIPTION
**Pull request by paulofelipefeitosa**
#### What is the purpose of this pull request?
This PR is part of the changes discussed in #111, it adds the implementation of the `aggregate_context_by_id` InputMaker function.

#### What problem is this solving?
The function reads a CSV file that contains triples in the format `id;epoch_timestamp;action`. Each triple represents a request identified by the `id` column, its arrival timestamp identified by the `epoch_timestamp` column, and its `action` (search or update).
Then, the function should aggregate the request's context (arrival timestamp and action) by their ids.

#### How should this be manually tested?

**Manual tests**
`python3 main.py --requests_context_file=<> --queries_file=<> --target_collection_name=<> --batch_groups_dir=<> --output_file=<>
`
**Unit tests**
`pip3 install requeriments.txt`
`python3 -m unittest input_maker_test.TestInputMaker`

#### Types of changes
- New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: replayer/input_maker/input_maker.py [hunk 1]
```diff
@@ -25,10 +25,34 @@ def process(req_context_filepath, queries_filepath, target_collection_name, batc
...
+        line_content = line[:-1].split(';')
+        if len(line_content) < 3:
+            raise Exception('line (%s) at file (%s) is incompleted, expected at least 3 columns separated by \';\'' % (line, req_context_filepath))
+        
+        try:
+            req_id = int(line_content[0])
+        except ValueError:
+            raise Exception('expected an integer value at column 0 from line (%s) at file (%s)' % (line, req_context_filepath))
+        try:
+            arrival_epoch = float(line_content[1].replace(',', '.'))
```
Comment (thiagomanel) [inline replayer/input_maker/input_maker.py]: Uhm ... this line 43 is odd. isn't expected to receive a string such as number1.number2? the replace parameter seems inverted.

File: replayer/input_maker/input_maker.py [hunk 2]
```diff
@@ -25,10 +25,34 @@ def process(req_context_filepath, queries_filepath, target_collection_name, batc
 
     return sort_replayer_requests(replayer_requests)
 
-
 def aggregate_context_by_req(req_context_filepath):
-    # TODO(): implement this method.
-    return defaultdict()
+    context_file = open(req_context_filepath, 'r')
```
Comment (thiagomanel) [inline replayer/input_maker/input_maker.py]: `with open(req_context_filepath) as context_file:` here?

**Comment by thiagomanel** [comment]
> Thanks, @paulofelipefeitosa. I think the code is mostly ok. I've added two questions, though.

a small complaint: I had to run `python3 -m pip install parameterized`. If the code grows more complex, it would be nice to have this automated (or described in a doc)

File: replayer/input_maker/input_maker.py (hunk 1 above)
Comment (thiagomanel) [reply replayer/input_maker/input_maker.py]: Btw, i guess you don’t need this line

File: replayer/input_maker/input_maker.py (hunk 2 above)
Comment (paulofelipefeitosa) [reply replayer/input_maker/input_maker.py]: Done.

File: replayer/input_maker/input_maker.py (hunk 1 above)
Comment (paulofelipefeitosa) [reply replayer/input_maker/input_maker.py]: You are right, we do not need to replace the string.
At some point will be necessary to transform this value into float to represent a _usec_  timestamp, I decided to do it here.

**Comment by paulofelipefeitosa** [comment]
Thanks for the review @thiagomanel, I added a requirements.txt file, we have a task to update the documentation and there will be included the steps to run the src code and the unit tests.

**Pull request merged**